### ISSUE TITLE
Making use of Motion and/or MotionEye for turning the camera into a motion sensor

### ISSUE THREAD

**Issue by AssetBurned** [Sunoo/homebridge-camera-ffmpeg#1084]
**Describe the solution you'd like:**
To my understanding this plugin requires an additional physical motion sensor to perform any actions. However there is an project called MotionEye (https://github.com/ccrisan/motioneye) that offers a web UI for the Motion (https://motion-project.github.io) that allows to interact with the camera stream and identifies any changes in the video stream and then performs actions. 

**Describe alternatives you've considered:**
- As projects such as https://github.com/Jpja/FFmpeg-Detect-Copy-Motion show it would also be possible to use FFmpeg to perform this task.
- looks like someone had already the idea to create a separated plugin https://github.com/SoulOnFire/homebridge-motioneye but it looks abandoned even before anything was really started.

**Additional context:**
Obviously it would save money and setup time to not having to deal with a separated movement sensor.

**Comment by Sunoo** [comment]
One user of this plugin wrote a guide for using Motion to generate the motion alerts: https://sunoo.github.io/homebridge-camera-ffmpeg/automation/motion.html

I've been intending to clean that document up for a bit, but haven't had the chance. I know some users have set this up with MotionEye, but I don't believe they've shared their configs for having motion flow through.

Is there something specific you are looking to be added to this plugin, or would it just be better documentation for using those together?

**Comment by AssetBurned** [comment]
hmmm i think this is a multilayer question you ask there.
The documentation could use some general revamp for a more general approach to this topic.

But what bugs me is that if homebridge, this plugin & motioneye all runs on the same raspberry, why is there a need to add an MQTT setup into the mix?

**Comment by NorthernMan54** [comment]
Personally I find that motioneye requires a dedicated rpi due to the cpu overhead when using it with multiple cameras.

**Comment by Sunoo** [comment]
When I rewrite it, I plan to do so using the HTTP service instead of MQTT, since as you said, that’s a bit easier. I have not used Motion myself though, so I can’t speak for much of it. When I have some time, I do plan to experiment with it, if for no other reason than to make sure whatever documentation I write is accurate.

**Comment by AssetBurned** [comment]
> Personally I find that motioneye requires a dedicated rpi due to the cpu overhead when using it with multiple cameras.

Yeah looking at video stream (not to mention the CPU load) is a bit scary. I have a test setup with an RP3B+ and an cheap chinese knock off camera. Still the CPU jumps up to 70% when i wave the hand in front of the camera. Nothing else than a basic Homebridge setup with just this plugin and MotionEye. the created video files are fine, but not the stream forwarded to the iPhone at least in the first few seconds.

If I would deploy this setup, would just dedicate one Pi per camera, and one of those Homebridge instances.

That is why i was mentioning in my original post FFmpeg, i think it also has some basic functionality for detecting if something major has changed in the frame.

**Comment by Sunoo** [comment]
I can look into the FFmpeg method when I have a chance (may be a bit, life is crazy right now), but I’d be surprised if it was meaningfully better for CPU usage than the other methods. It’d still need to be an extra instance of FFmpeg running at all times, and it would need to be doing some amount of analysis of each frame as they come in.

**Comment by NorthernMan54** [comment]
@AssetBurned Personally after trying out MotionEye for a few weeks, and seeing how it improved the usability of my Wyze and Eufy cams, I bit the bullet and dedicated a RPI 3B+ for 2 camera's.  I went from flakiness with the wye cams running the RTSP firmware to rock solid reliability, with the added bonus of local storage of motion events.  It also simplified management of them, as I now just use the MotionEye console rather than multiple vendor apps.

To get the CPU manageable, I did tune the config and reduced the frame rate of the web console and other tweaks and reduced any transcoding on the RPI.

PS Here is my CPU load ( 2 is 50% cpu, 3 is 75% cpu )

![image](https://user-images.githubusercontent.com/19808920/112376806-48051e00-8cbb-11eb-931d-554134f142b8.png)
